app/src/main/java/swen_anigans/mathematicfanatic/ClassroomActivity.java

```diff
@@ -168,21 +150,18 @@ public void goToLocker(View view){
     }
 
     public void checkAnswer(View view){
-        EditText answerInput = (EditText) findViewById(R.id.editClassroomAnswer);
-        String answer = answerInput.getText().toString();
-        CharSequence text;
-        if (!answer.isEmpty()) {
-            int quizAnswer = Integer.parseInt(answer);
-            answers.set(pageNumber-1, quizAnswer);
-            boolean correct;
-            if(quizAnswer == expectedAnswers.get(pageNumber - 1)){
-                text = "Correct!";
+        TextView answerInput = (TextView) findViewById(R.id.editClassroomAnswer);
+        String answerText = answerInput.getText().toString();
+        String text;
+        if(!answerText.isEmpty()){
```

Comment:
We could have it so that you save the answer before doing anything else in check answer. 

so it could be like:
```
saveAnswer();
if(questions.get(pageNumber-1).submittedAnswer == 0)
{
    text = "please answer an answer first"
}
else if(questions.get(pageNumber -1).checkAnswer())
{
     text = Correct
}
`else`
{
    text = Incorrect
}
```